cksum: permit repeated flags, handle overrides correctly, implement base64 output

BenWiederhake opened the pull request:
This PR changes and tests three things:
- Accept `--tag` and `--untagged` in any order, use the last occurrence.
- Accept repetitions of everything, flags and arguments like `--algorithm`.
- Accept and correctly handle `--base64` flag.

All three of these are GNU behavior bugs, i.e. GNU does it differently, and uutils wants to handle it like GNU does.

This is work towards #5998.

Note that `cksum` is still woefully incomplete; most prominently, the `-c` option is missing.

sylvestre commented on src/uu/cksum/src/cksum.rs:
```diff
@@ -186,17 +194,33 @@ where
+                        sum_hex
+                    }
+                    _ => {
+                        //data_encoding::BASE64.encode(decode(sum_hex).unwrap())
```
could you please remove this comment ?

BenWiederhake replied:
Oh! Sorry, I must have been more tired than I thought. Thank you!

BenWiederhake commented:
Changes since last push:
- Removed commented-out test code. Thanks @sylvestre for spotting it!
- Double-checked all the other commits
- Rebased on current `main`

BenWiederhake commented:
Only now I see that a similar, but older PR from @tertsdiepraam exists: #5122
I'm not sure what's a good way forward.

tertsdiepraam commented:
Yours can take priority. I haven't looked at mine in a while and I can always rebase it.

cakebaker commented on tests/by-util/test_cksum.rs:
```diff
@@ -291,6 +325,20 @@ fn test_length_is_zero() {
+#[test]
+fn test_length_repeated() {
+    new_ucmd!()
+        .arg("--length=10")
+        .arg("--length=123456")
+        .arg("--length=0")
```
I don't know if you have chosen the invalid values `10` and `123456` on purpose. At least they helped me discover a consistency bug in GNU `cksum`: `--length=10 --length=0` shows an error whereas `--length=123456 --length=0` doesn't. They fixed it to match our implementation and no longer show an error in such a case. So we can leave this test as it is.

Thanks for the example :)

BenWiederhake replied:
That was not intentional, but I guess it goes to show that I should really run *every single little command* also with the GNU implementation. :D

BenWiederhake commented:
Changes since last push: Just rebased, to demonstrate that it's still working.

Is there anything missing?
